src/librustc_trans/trans/expr.rs

```diff
@@ -2477,11 +2507,144 @@ fn with_overflow_check<'a, 'b>(bcx: Block<'a, 'b>, oop: OverflowOp, info: NodeId
                     InternedString::new("arithmetic operation overflowed")));
 
         (bcx, result)
+    }
+}
+
+impl OverflowOpViaInputCheck {
+    fn build_with_input_check<'blk, 'tcx>(&self, bcx: Block<'blk, 'tcx>,
+                                          info: NodeIdAndSpan,
+                                          lhs_t: Ty<'tcx>, lhs: ValueRef,
+                                          rhs_t: Ty<'tcx>, rhs: ValueRef,
+                                          binop_debug_loc: DebugLoc)
+                                          -> (Block<'blk, 'tcx>, ValueRef)
+    {
+        let to_llty = |r| bcx.ccx().tn().type_to_string(Type::from_ref(r));
+        {
+            let given_llty = type_of::type_of(bcx.ccx(), rhs_t);
```

Comment:
ah, this whole sub-block is debugging code that is no longer needed; I will remove it.